rd-net/RdFramework/Impl/SocketWire.cs

```diff
@@ -224,8 +260,23 @@ private int ReceiveFromPkgBuffer(byte[] buffer, int offset, int size)
             Int32 len = UnsafeReader.ReadInt32FromBytes(myPkgHeaderBuffer.Data);
             Int64 seqN = UnsafeReader.ReadInt64FromBytes(myPkgHeaderBuffer.Data, sizeof(Int32));
 
-            if (len == ACK_MSG_LEN) 
+            if (len == PING_LEN)
+            {
+              myPonger.SendAsync(Unit.Instance);
+              continue;
+            } 
+            if (len == PONG_LEN)
```

Comment:
Why do we need PONG events at all? Why PINGs aren't enough?

Reply:
There could be different heartbeat intervals on client and server sides, so it wouldn't be enough to react only on PINGs. Also what's about multiclient connection at all?

Reply:
Okay, let it be the same intervals. Suppose the case when we lose outgoing messages, but successfully accept incoming messages. How to determine whether the connection is stable or not without reactions on PINGs?

Reply:
How can this happen? Could you explain the example?

Reply:
It can be happen due to network congestion only in one direction or firewall or smth like that. Are we going to exclude these cases from consideration?

Reply:
This may be a valid point even if it's a corner case. But I still think we can use only one type of message with fully asynchronous interaction. Just put two additional fields in this message: timestamp of last received message from counterpart and your timestamp.